src/actions/mempool-api.ts

```diff
@@ -0,0 +1,61 @@
+"use server";
+import { getUtxosBitcoinDaemon } from "@/app/api/proxy/[...proxy]/rpc-handler-core";
+import { RateLimiter } from "limiter";
+
+class LimiterLibraryRateLimiter {
+  maxRequestWindowMS: number = 60 * 1000;
+  limiter: RateLimiter;
+  constructor() {
+    this.limiter = new RateLimiter({
+      tokensPerInterval: 500,
+      interval: "minute",
+      fireImmediately: false,
+    });
+  }
+
+  async acquireToken(fn: () => Promise<Response>): Promise<Response> {
+    if (this.limiter.tryRemoveTokens(1)) {
+      await nextTick();
+      return fn();
+    } else {
+      await sleep(this.maxRequestWindowMS);
+      return this.acquireToken(fn);
+    }
+  }
+}
+
+const factory = <T extends (...args: any[]) => Promise<Response>>(fn: T) => {
+  const tokenBucket = new LimiterLibraryRateLimiter();
+  return (...args: Parameters<T>) => {
+    return fetchAndRetryIfNecessary(() =>
+      tokenBucket.acquireToken(() => fn(...args)),
+    );
+  };
+};
+export const mempoolFetch = factory((...args: Parameters<typeof fetch>) =>
+  fetch(...args),
+);
```

Comment:
I'm not sure if we want to use the retry version for everything or just for the tx fetching when creating the emily request. With a bunch of those tasks running do you know if we risk some form of resource exhaustion?

Reply:
Good point